lib/BackgroundJob/BackgroundScanner.php

```diff
@@ -243,6 +287,20 @@ protected function getToRescanFiles() {
 		return $qb->execute();
 	}
 
+	protected function getOutdatedFiles() {
+		// We do not want to keep scanning the same files. So only scan them once a day at most.
+		$yesterday = time() - (24 * 60 * 60);
+
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('fileid')
+			->from('files_antivirus')
+			->andWhere($qb->expr()->lt('check_time', $qb->createNamedParameter($yesterday)))
+			->orderBy('check_time', 'DESC')
```

Comment:
```suggestion
			->orderBy('check_time', 'ASC')
```

asc means getting higher, so you would actually start with the oldest files, instead of rescanning the newest files over and over again.

Reply:
I knew I should not code after a busy day